### PR TITLE
cloudhub/udsserver: add checks before using the unix addr

### PR DESCRIPTION
**Pull request by Iceber**
Signed-off-by: Iceber Gu <wei.cai-nat@daocloud.io>

**What type of PR is this?**

/kind bug
/kind feature

**What this PR does / why we need it**:
Better handling of the two cases `"UNIX:///tmp/uds.sock"` and `"unix://    "`

Add more checks before using the addr
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/cloudhub/servers/udsserver/uds.go [hunk 1]
```diff
@@ -54,7 +57,7 @@ func (us *UnixDomainSocket) StartServer() error {
 		return err
 	}
 	if proto == "unix" {
-		addr = "/" + addr
+		addr = filepath.Join("/" + addr)
 		if err := os.Remove(addr); err != nil && !os.IsNotExist(err) { //nolint: vetshadow
```
Comment (Iceber) [inline cloud/pkg/cloudhub/servers/udsserver/uds.go:61]: This is a very dangerous operation

Comment (Iceber) [reply cloud/pkg/cloudhub/servers/udsserver/uds.go]: Add more checks before removing

**Comment by Iceber** [comment]
@fisherxu PTAL

File: cloud/pkg/cloudhub/servers/udsserver/uds.go [hunk 2]
```diff
@@ -54,11 +57,12 @@ func (us *UnixDomainSocket) StartServer() error {
...
-		addr = "/" + addr
-		if err := os.Remove(addr); err != nil && !os.IsNotExist(err) { //nolint: vetshadow
-			klog.Errorf("failed to remove addr: %v", err)
+		addr = filepath.Join("/" + addr)
+		if err := checkUnixSocket(addr); err != nil {
+			klog.Errorf("failed to check unix socket addr: %v", err)
 			return err
 		}
+		defer os.Remove(addr)
 	}
```
Comment (gy95) [inline cloud/pkg/cloudhub/servers/udsserver/uds.go:66]: i think don't need this,  function `checkUnixSocket` already remove socket. or we can remove `os.Remove` from `checkUniSocket` and exec `os.Remove` out of `checkUniSocket` and return error to `StartServer`

Comment (Iceber) [reply cloud/pkg/cloudhub/servers/udsserver/uds.go:66]: This is to ensure that the created sockets are removed when the unix server exits，although for the time being it does not appear that the `StartServer` will exit gracefully(TODO)

It is not necessary to have a socket when the program is not running.

Comment (gy95) [reply cloud/pkg/cloudhub/servers/udsserver/uds.go:66]: got it, thanks

**Comment by Iceber** [comment]
@GssC PTAL, Thanks

**Pull request closed without merging**